Core/Core/Router/Router.swift

```diff
@@ -89,32 +86,33 @@ public extension RouterProtocol {
 
         if let displayModeButton = from.displayModeButtonItem,
             from.splitViewController?.isCollapsed == false,
-            options.detail || from.isInSplitViewDetail,
-            options.modal == nil {
+            options.isDetail || from.isInSplitViewDetail,
+            !options.isModal {
             view.addNavigationButton(displayModeButton, side: .left)
             view.navigationItem.leftItemsSupplementBackButton = true
         }
 
-        if options.addDoneButton {
-            view.addDoneButton(side: .left)
-        }
         var nav: UINavigationController?
         if options.embedInNav {
             nav = view as? UINavigationController ?? UINavigationController(rootViewController: view)
         }
 
-        if let modal = options.modal {
+        switch options {
+        case let .modal(modalOptions):
+            if modalOptions.addDoneButton {
+                view.addDoneButton(side: .left)
+            }
             nav?.navigationBar.useModalStyle()
-            if let presentationStyle = modal.presentationStyle {
+            if let presentationStyle = modalOptions.0 {
                 (nav ?? view).modalPresentationStyle = presentationStyle
             }
-            if #available(iOS 13, *), modal.inPresentation {
+            if #available(iOS 13, *), !modalOptions.isDismissable {
                 (nav ?? view).isModalInPresentation = true
             }
             from.present(nav ?? view, animated: true, completion: completion)
-        } else if from.splitViewController != nil, options.detail, !from.isInSplitViewDetail {
+        case .detail where from.splitViewController != nil  && !from.isInSplitViewDetail:
             from.showDetailViewController(nav ?? view, sender: from)
-        } else {
+        default:
```

Comment:
I'd prefer to have `case .push:` here so if we added something else the compiler would force us to handle it.

Reply:
It's not quite a case .push though. It's also the .details where the "where" clause failed

Reply:
case .detail, .push: then?